per/ops_learning_summary.py

```diff
@@ -0,0 +1,818 @@
+        "so that they are supported by evidence coming from more than one report, "
+        "and there is ONLY ONE entry per subtype. Always integrate in the paragraph evidence that supports "
+        "it from the data available from multiples reports or items, include year and country of the evidence. "
+        "The length of each paragraph MUST be between 20 and 30 words."
+        " Important:"
+        "- ONLY create one summary per subtype"
+        "- DO NOT mention the ids of the excerpts in the content of the summary."
+        "- DO NOT use data from any source other than the one provided. "
+        "Output Format:"
+        "Provide your answer in valid JSON form. Reply with ONLY the answer in JSON form and include NO OTHER COMMENTARY."
+        '{"0": {"type": "sector", "subtype": "shelter", "excerpts id":"43, 1375, 14543", "content": "lorem ipsum"}, '
+        '"1": {"type": "component", "subtype": "Information Management", "excerpts id":"23, 235", "content": "lorem ipsum"}, '
+        '"2": {"type": "sector", "subtype": "WASH", "excerpts id":"30, 40",  "content": "lorem ipsum"}}'
+    )
+
+    system_message = (
+        "# CONTEXT # I want to summarize a set of lessons learned from a set of past emergency response operations "
+        "to extract the most useful and actionable insights."
+        "# STYLE # Use a writing style that is professional but informal."
+        "# TONE # Encouraging and motivating."
+        "# AUDIENCE # The audience is emergency response personnel from the Red Cross and Red Crescent. "
+        "They are action-oriented people who have very little time so they need concise, "
+        "not obvious information that can be easily consumed and acted upon in the time of a response."
+    )
+
+    primary_instruction_prompt = (
+        "You should:"
+        "1. Describe, Summarize and Compare: Identify and detail the who, what, where and when"
+        "2. Explain and Connect: Analyze why events happened and how they are related"
+        "3. Identify gaps: Assess what data is available, what is missing and potential biases"
+        "4. Identify key messages: Determine important stories and signals hidden in the data"
+        "5. Select top three: Select up to three findings to report"
+    )
+
+    secondary_instruction_prompt = (
+        "You should for each section in the data (TYPE & SUBTYPE combination):"
+        "1. Describe, Summarize and Compare: Identify and detail the who, what, where and when"
+        "2. Explain and Connect: Analyze why events happened and how they are related"
+        "3. Identify gaps: Assess what data is available, what is missing and potential biases"
+        "4. Identify key messages: Determine if there are important stories and signals hidden in the data"
+        "5. Conclude and make your case"
+    )
+
+    client = AzureOpenAI(
+        azure_endpoint=settings.AZURE_OPENAI_ENDPOINT, api_key=settings.AZURE_OPENAI_KEY, api_version="2023-05-15"
+    )
```

Comment:
Use cached_property for this

Add a `TODO: to create a separate object for integration with Open AI for common use cases`